provider/pkg/await/await.go

```diff
@@ -386,14 +397,14 @@ func Update(c UpdateConfig) (*unstructured.Unstructured, error) {
 	// Wait until patch resolves as success or error. Note that the conditional is set up to log only
 	// if we don't have an entry for the resource type; in the event that we do, but the await logic
 	// is blank, simply do nothing instead of logging.
-	id := fmt.Sprintf("%s/%s", c.Inputs.GetAPIVersion(), c.Inputs.GetKind())
+	id := fmt.Sprintf("%s/%s", currentOutputs.GetAPIVersion(), currentOutputs.GetKind())
 	a := awaiters
 	if c.awaiters != nil {
 		a = c.awaiters
 	}
 	if awaiter, exists := a[id]; exists {
-		if metadata.SkipAwaitLogic(c.Inputs) {
-			logger.V(1).Infof("Skipping await logic for %v", c.Inputs.GetName())
+		if metadata.SkipAwaitLogic(currentOutputs) {
```

Comment:
Should we check the inputs for this instead? There might be edge cases where a mutating webhook could override/delete this annotation. In that case, it might be preferred to stick to the user's intention of skipping the await?

Reply:
I think the counter-argument is equally true, that someone set `skipAwait` using `kubectl edit` and are confused about why it is ignored (e.g. in an update case).  We chose to model the parameters as annotations, I feel like we should lean into that.

Also, in `Delete`, the `skipAwait` annotation was drawn from the outputs even before this PR.

WDYT?

Reply:
"If the annotation is removed via `kubectl edit`, wouldn't the subsequent `pulumi up` operation still reintroduce those `skipAwait` annotations since they are defined in the user's program (ie. drift correct)? It appears that the `pulumi.com/skipAwait` annotation is closely associated with the specific `pulumi up` operation due to this.

Referring to the discussion in https://github.com/pulumi/pulumi-kubernetes/issues/179#issuecomment-462506076, it appears that the decision to utilize annotations instead of an SDK flag was primarily for convenience and to ensure compatibility with specific objects within a given `yaml.ConfigGroup` resource for example.

Utilizing the annotation from `inputs` appears more logical to me, particularly since Pulumi is a CLI tool, thus leaving the decision and configuration of whether to wait for an object to be in a ready state up to the user executing the command.

> Also, in Delete, the skipAwait annotation was drawn from the outputs even before this PR.

It's possible this was an oversight in our codebase. However, in most scenarios, encountering this issue is likely rare, as the scenario of a mutating webhook emptying annotations is probably uncommon in practice.

Another pertinent example is the `pulumi.com/patchForce` annotation, which is closely linked to inputs as well, further supporting the rationale for using inputs for logic depending on annotation options for Pulumi resources."

Reply:
It is true that the pulumi annotations seem like "meta-annotations" in that they inform behavior (like an option) but aren't really intended to be applied as an annotation to the object. In the case of `patchForce` it leads to problems.  I've advocated elsewhere for the pulumi annotations to be dropped from the actual patch for this reason.

I think you've convinced me to use the input, and will adjust `Delete` too.